docs/conceptual/runtime/weight.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Transaction Weight
+---
+
+When transactions are submitted to a blockchain, they are executed by the nodes
+in the network. To be economically sustainable, nodes charge a fee to execute a
+transaction that must be covered by the submitter of the transaction. The cost
+to execute transactions could vary over orders of magnitude, and thus Substrate
+provides a flexible mechanism for characterizing the total, presumably minimum,
+cost of a transaction in order to be included into a block.
+
+Furthermore, a number of resources in any chain can be limited by definition
+such as time, memory and computation. A mechanism should exist to prevent
+individual resource-consuming components of the chain, including but not limited
+to dispatchables, to consume too much of any of the mentioned resources. This
+concept is represented in substrate by weights.
+
+## Transaction Weight
+
+The fee system is dependent on the
+[weight](https://crates.parity.io/sr_primitives/weights/index.html) system.
+Hence, we will first briefly describe how weights work.
+
+As mentioned, weights can technically refer to, or represent, numerous _limited_
+resources. A custom implementation may use complex structures to demonstrate
+this. At the time of this writing, substrate weights are simply a [numeric
+value](https://crates.parity.io/sr_primitives/weights/type.Weight.html). Each
+dispatchable function is given a weight using the `#[weight = $x]` annotation,
+where `$x` is some function capable of determining the weight of the dispatch.
+`$x` can access and examine the arguments of the dispatch. Nonetheless, a weight
+calculation should always:
+- Be computable __ahead of dispatch__. A block producer, or a validator, should
+  be able to examine the weight of a dispatchable before actually deciding to
+  accept it or not.
+- While not enforced, calculating the weight should not consume much resource
+  itself. It does not make much sense to spend almost the same amount of time as
+  dispatching a transaction to estimate how much weight it will consume. Thus,
+  weight computation should typically be very fast, much faster than
+  dispatching. This is analogous to a `poll()` function of the `Future` trait.
+  It really doesn't make sense to poll a future if it is going to stall us until
+  the future itself is resolved; A sensible `poll` must return fast.
+- Finally, if a transaction is very variable in the amount of resources it
+  consumes, it should manually prevent restrict certain dispatches through any
+  means that makes sense to that particular chain. An example of such dispatches
+  in substrate is the call to a smart contract (todo: link when this is
+  implemented).
```

Comment:
That sounds like the right thing to do. Although, I still consider the "next" docs to be a roughish playground.

I am not against merging to them with a TODO, and then doing a final pass to clean up TODOs before we make it the release version of our docs.

I have been doing this with the NEXT STEPS because its hard to know all the things we can reference to, until we write all the docs and make recipes and stuff.

Reply:
Let's put comments inside `<!--html comment tags-->` so that we can search for them but they don't show up on the site.